### ISSUE TITLE
Electron and Bytenode: Is there a way to make a stronger encryption?

### ISSUE THREAD

**Issue by plamendobrev** [OsamaAbbas/bytenode#64]
So, I just learned that I have to compile my .js file to .jsc using my Electron app itself. I did that but there's a slight difference between doing it from the console and the app. The very reason I'm doing this is to hide my database information and when I do it from the app, it's technically still exposed. Is there a way to hide it using V8 Bytenode or my only option is to migrate to NW.js or just use another method?

```
const bytenode = require('bytenode');
bytenode.runBytecodeFile('./login.jsc');
```

**Comment by s100** [comment]
I don't believe `bytenode` is intended to be a form of encryption. Though not technically public or documented, the V8 bytenode format is not secret or intentionally obfuscated, and it can be very easily interpreted using a V8 interpreter. You may need to change the architecture of your application so that your database information isn't part of it.

**Comment by OsamaAbbas** [comment]
There are a bit of confusion here. When we say that you have to use Electron to generate the `.jsc` files, what really happens is that the embedded Node inside Electron will generate them. So, in terms of "security" they are the same. In fact, if you have turned off the "sanity checks" from Node.js source code, them you will be able to run any `.jsc` files given that they have the same format (i.e. the same major v8 version in most cases). But as we want to make bytenode usable with the already-built-binaries, we have not used the previous approach, because no one will prefer using our custom binary. (It is more difficult to maintain too).

That being said, I have to say that even if you are using c or c++, and you will compile your code amd hide your secret information inside the binary compiled program: this is very bad and weak archeticture. Your information is not safe nor secured at all, even if you use the known tricks to hide strings inside the binaries (google them and you will find many).

Bytenode was created mainly to help developers hide their code if they don't feel comfortable with the idea that their client will have the plain source code for whatever reason. Or if they want to make it difficult for their smart client (or competitor) to copy their work. Especially in places and countries that don't have the legal base to prevent such practices.

In other words, Bytenode is an alternative to the correct and sound legal approach to prevent software theft. It is NOT meant to securely protect any information inside the code. Even machine language is NOT safe nor secure for hiding such information.

**Comment by OsamaAbbas** [comment]
> when I do it from the app, it's technically still exposed

How is it still exposed? Could you explain that please?
